### PR TITLE
[docs]: Fix the documentation for Icon Component

### PR DESCRIPTION
**Pull request by sikehish**
This PR is for issue #10972 

**Summary of Changes:**
- Replaced "widget" with "component" wherever applicable (without modifying URLs).
- Updated "Fx" to "fx" (bold, lowercase, no backticks).
- Removed the 24px padding-bottom for all `##` headers and ensured only padding-top: 24px is applied.
- Updated descriptions for "On hover" and "On click" events based on the reference.
- Added a full stop at the end of sentences in the event description column.
- Inserted a divider (`---`) before the "Styles" section.

**Files Updated:**
- ToolJet/docs/docs/widgets/icon.md
- ToolJet/docs/versioned_docs/version-2.50.0-LTS/widgets/icon.md

### CODE REVIEW

File: docs/docs/widgets/icon.md [hunk 1]
```diff
@@ -36,58 +34,60 @@ Check [Action Reference](/docs/category/actions-reference) docs to get the detai
...
+---
 
 ## Styles
 
-| <div style={{ width:"100px"}}> Style </div> | <div style={{ width:"100px"}}> Description </div> | <div style={{ width:"135px"}}> Expected Value </div> |
-|:----------- |:----------- |:------------- |
-| Icon color |  You can change the color of the Icon widget by entering the Hex color code or choosing a color of your choice from the color picker. |
-| Visibility | This is to control the visibility of the widget. | If `{{false}}` the widget will not visible after the app is deployed. | It can only have boolean values i.e. either `{{true}}` or `{{false}}`. By default, it's set to `{{true}}`. |
-| Box shadow | This property adds a shadow to the widget. | You can use different values for box shadow property like offsets, blur, spread, and the color code. |
+| <div style={{ width:"100px"}}> Style </div> | <div style={{ width:"100px"}}> Description </div>                                                                                       | <div style={{ width:"135px"}}> Expected Value </div>                                                 |
```
Comment (RudraPratik30) [inline docs/docs/widgets/icon.md:83]: Hey @sikehish can you please revert all the extra spacing and dashes you have added? For some reason they are breaking on site. Check it on [render](https://tooljet-pr-11002.onrender.com/docs/beta/widgets/icon).

image

File: docs/docs/widgets/icon.md [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
 
 </div>
 
-<div style={{paddingTop:'24px', paddingBottom:'24px'}}>
+<div style={{paddingTop:'24px'}}>
 
 ## Events
 
-To add an event, click on the Icon widget's configuration handle to open the widget properties on the components drawer on the right. Go to the **Events** section and click on **+ Add handler**.
+The Icon component supports the following events:
```
Comment (RudraPratik30) [inline docs/docs/widgets/icon.md:22]: Please remove this line as well.

**Review by RudraPratik30 — CHANGES_REQUESTED** [review]
Hey @sikehish please make the mentioned changes in both versions.

**Comment by sikehish** [comment]
Sure, I'm looking into it.

File: docs/docs/widgets/icon.md (hunk 1 above)
Comment (sikehish) [reply docs/docs/widgets/icon.md]: I appolgize for this. I actually had not added any extra spacing and dashes. There was some issue with auto formatting. I have reverted it.

**Comment by sikehish** [comment]
@RudraPratik30 I have made the necessary changes. Do let me know if anything else needs to be changed.

**Review by RudraPratik30 — APPROVED** [review]
Hey @sikehish, thanks for contributing!
The changes look good to me and forwarding it to the team for final review.
You can expect your PR to be merged by the end of next week, thanks again!

**Comment by sikehish** [comment]
> Hey @sikehish, thanks for contributing! The changes look good to me and forwarding it to the team for final review. You can expect your PR to be merged by the end of next week, thanks again!

Sounds great! Thank you too for the oppurtunity :)

**Review by karan-rathod-316 — APPROVED** [review]
Looks good to me!

**Review by johnsoncherian — APPROVED** [review]

**Pull request merged**